src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
```

Comment:
your version is definitely more explicit, but let's delegate some of the logic to underscore for clarity. The for-loop-based check is a pretty common pattern, so I think it is safe to replace that with the `filter` method. What we can do is to add a few sentences in between the chain so that it becomes more obvious for future developers taking a look at the code.